src/library/itunes/itunesmacosimporter.mm

```diff
@@ -0,0 +1,282 @@
+        //
+        //     House
+        //     House #2
+        //     House #3
+        //
+
+        // Avoid empty playlist names
+        if (name.isEmpty()) {
+            name = "(empty)";
+        }
+
+        auto existing = m_playlistDuplicatesByName.find(name);
+        if (existing != m_playlistDuplicatesByName.end()) {
+            m_playlistDuplicatesByName[name] += 1;
+            return QString("%1 #%2").arg(name).arg(
+                    m_playlistDuplicatesByName[name] + 1);
+        } else {
+            m_playlistDuplicatesByName[name] = 0;
+            return name;
+        }
+    }
+
+    bool isPlaylistShown(ITLibPlaylist* playlist) {
+        // Filter out the primary playlist (since we already show the library
+        // under the main iTunes node)
+        bool isPrimary;
+        if (@available(macOS 12.0, *)) {
+            isPrimary = [playlist isPrimary];
+        } else {
+            isPrimary = [playlist isMaster];
+        }
+        if (isPrimary) {
+            return false;
+        }
+
+        // Filter out automatically generated 'category' playlists,
+        // most of these are either redundant (e.g. the 'Music' playlist)
+        // or include non-audio content.
+        switch (playlist.distinguishedKind) {
+        case ITLibDistinguishedPlaylistKindMusic:
+        case ITLibDistinguishedPlaylistKindMovies:
+        case ITLibDistinguishedPlaylistKindPodcasts:
+        case ITLibDistinguishedPlaylistKindAudiobooks:
+        case ITLibDistinguishedPlaylistKindTVShows:
+        case ITLibDistinguishedPlaylistKindiTunesU:
+        case ITLibDistinguishedPlaylistKindMusicVideos:
+        case ITLibDistinguishedPlaylistKindLibraryMusicVideos:
+        case ITLibDistinguishedPlaylistKindHomeVideos:
+        case ITLibDistinguishedPlaylistKindApplications:
+        case ITLibDistinguishedPlaylistKindMusicShowsAndMovies:
+            return false;
+        default:
+            break;
+        }
+
+        return true;
+    }
+
+    void importPlaylist(ITLibPlaylist* playlist,
+            QSqlQuery& queryInsertToPlaylists,
+            QSqlQuery& queryInsertToPlaylistTracks) {
+        if (!isPlaylistShown(playlist)) {
+            return;
+        }
+
+        int playlistId = dbIdFromPersistentId(playlist.persistentID);
+        int parentId = playlist.parentID
+                ? dbIdFromPersistentId(playlist.parentID)
+                : -1;
+
+        QString playlistName = uniquifyPlaylistName(qStringFrom(playlist.name));
+
+        queryInsertToPlaylists.bindValue(":id", playlistId);
+        queryInsertToPlaylists.bindValue(":name", playlistName);
+
+        if (!queryInsertToPlaylists.exec()) {
+            LOG_FAILED_QUERY(queryInsertToPlaylists);
+            return;
+        }
+
+        m_playlistNameByDbId.insert(playlistId, playlistName);
+        m_playlistChildsByDbId.insert(parentId, playlistId);
+
+        int i = 0;
+        for (ITLibMediaItem* item in playlist.items) {
```

Comment:
Makes sense, can you add that rationale as a comment above the loop?
```suggestion
        // We use an Obj-C for-in loop here since the underlying collecting
        // is also an Obj-C type. 
        for (ITLibMediaItem* item in playlist.items) {
```

Reply:
We could do that, though I would _generally_ try to abstain from these kinds of comments that only document syntax (I'm assuming the reader knows basic Objective-C, e.g. we don't document what a C++ for-loop is either).

If the reader sees `for-in` in a `.m` or `.mm` file it should be clear that we are dealing with Objective-C types.

Reply:
I agree in the general case, though I argue that we explicitly can't assume that the reader has knowledge that in depth. Neither the article you linked, nor the 70 page pdf I found mention these pitfalls. As I said, nobody from the mixxx team knows Obj-C (nor do I feel like they have a great desire to learn it). Thats why I would prefer to treat the Obj-C code like language tutorial for people that are looking at this language for the first time. The really basic stuff can be assumed to be known because its the first thing you read in every article (for example invoking selectors) but the intermediate to advanced stuff would preferably be explained.

Reply:
Yeah, but isn't `for-in` pretty self-explanatory?

Reply:
Yes the concept is, but its not self-explanatory why we prefer a Obj-C for-in _over a C++ ranged-based-for_ in this particular case.

Reply:
I've added a quick note on this in 792f1b51bd38aecccdb11864d9928ffb3b0c6559